simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager

```diff
@@ -65,39 +240,58 @@ class RouteManager():
             rospy.logerr("Route mode '%s' unknown, exiting route manager", self.route_mode)
             return
 
-        poses = rospy.get_param('~poses')
-        if not poses:
+        poses = rospy.get_param('~poses', [])
+        if not poses and self.route_mode!='dynamic':
             rospy.loginfo("Route manager initialized no goals, unable to route")
-
+        
         self.goals = RouteManager.route_modes[self.route_mode](poses)
-        rospy.loginfo("Route manager initialized with %s goals in %s mode", len(poses), self.route_mode)
+        rospy.loginfo("Route manager initialized in %s mode", self.route_mode)
+
+        self.bad_goal_counter = 0
+
 
     def to_move_goal(self, pose):
+        if pose is None:
+            raise ValueError("Goal position cannot be NULL")
+
         goal = MoveBaseGoal()
         goal.target_pose.header.stamp = rospy.Time.now()
         goal.target_pose.header.frame_id = "map"
         goal.target_pose.pose.position = Point(**pose['pose']['position'])
         goal.target_pose.pose.orientation = Quaternion(**pose['pose']['orientation'])
         return goal
 
+
     def route_forever(self):
         rate = rospy.Rate(1)
         while not rospy.is_shutdown():
-            try:
-                rospy.loginfo("Route mode is '%s', getting next goal", self.route_mode)
-                current_goal = self.to_move_goal(next(self.goals))
-                rospy.loginfo("Sending target goal: %s", current_goal)
-                self.client.send_goal(current_goal)
-
-                if not self.client.wait_for_result():
-                    rospy.logerr("Move server not ready, will try again...")
-                else:
-                    if self.client.get_result():
-                        rospy.loginfo("Goal done: %s", current_goal)
-                rate.sleep()
-            except StopIteration:
-                rospy.loginfo("No goals, stopping route manager")
+            if self.bad_goal_counter > 10:
+                rospy.loginfo("Stopping route manager due to too many bad goals. Check that your occupancy map has 'Trinary' value representation and is not visually noisy/incorrect")
                 return
+            else:
+                try:
```

Comment:
I think having nested `try-except` blocks is confusing. Either combine them into one, or have one only encapsulate the call to `to_move_goal()` and the other only encapsulate `wait_for_message()`.